atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1424,23 +1424,76 @@ public void createTable(final TableReference tableRef, final byte[] tableMetadat
+        // additionally it is possible that this metadata-only update performs a schema mutation
+        // by altering the CFDef (ex. user changes metadata of existing table to have new compression block size)
+        // This does not require the schema mutation lock, however, as it does not alter the CfId
+        internalPutMetadataForTables(tablesToUpdateMetadataFor, putMetadataWillNeedASchemaChange);
     }
 
-    private void createTablesInternal(final Map<TableReference, byte[]> tableNamesToTableMetadata) throws Exception {
-        try {
-            clientPool.runWithRetry((FunctionCheckedException<Client, Void, Exception>) client -> {
-                KsDef ks = client.describe_keyspace(configManager.getConfig().keyspace());
-                Set<TableReference> existingTablesLowerCased = Sets.newHashSet();
+    private Set<TableReference> getExistingTablesLowerCased(Client client) throws TException {
+        KsDef ks = client.describe_keyspace(configManager.getConfig().keyspace());
+        Set<TableReference> existingTablesLowerCased = Sets.newHashSet();
 
-                for (CfDef cf : ks.getCf_defs()) {
-                    existingTablesLowerCased.add(fromInternalTableName(cf.getName().toLowerCase()));
+        for (CfDef cf : ks.getCf_defs()) {
+            existingTablesLowerCased.add(fromInternalTableName(cf.getName().toLowerCase()));
+        }
+        return existingTablesLowerCased;
+    }
+
+    private Map<TableReference, byte[]> filterOutNoOpMetadataChanges(
+            final Map<TableReference, byte[]> tableNamesToTableMetadata) {
+        Map<TableReference, byte[]> existingTableMetadata = getMetadataForTables();
+        Map<TableReference, byte[]> tableMetadataUpdates = Maps.newHashMap();
+
+        for (Entry<TableReference, byte[]> entry : tableNamesToTableMetadata.entrySet()) {
+            TableReference tableReference = entry.getKey();
+            byte[] newMetadata = entry.getValue();
+
+            // if no existing table or if existing table's metadata is different
+            if (!Arrays.equals(existingTableMetadata.get(tableReference), newMetadata)) {
+                Set<TableReference> matchingTables = Sets.filter(existingTableMetadata.keySet(), existingTableRef ->
+                        existingTableRef.getTablename().equalsIgnoreCase(tableReference.getTablename()));
+
+                // completely new table, not an update
+                if (matchingTables.isEmpty()) {
+                    tableMetadataUpdates.put(tableReference, newMetadata);
                 }
 
+                // table already exists but has different metadata, so we should perform an update
+                if (!matchingTables.isEmpty() && !Arrays.equals(
+                        existingTableMetadata.get(Iterables.getOnlyElement(matchingTables)), newMetadata)) {
+                    tableMetadataUpdates.put(tableReference, newMetadata);
+                }
```

Comment:
Sure, we can log something here along the lines of "table exists, but had the same metadata already, so we don't need to do anything"